apps/opik-backend/src/main/java/com/comet/opik/domain/DatasetItemDAO.java

```diff
@@ -464,37 +495,90 @@ private String getOrDefault(JsonNode jsonNode) {
-                        .orElse(null))
-                .spanId(Optional.ofNullable(row.get("span_id", String.class))
-                        .filter(s -> !s.isBlank())
-                        .map(UUID::fromString)
-                        .orElse(null))
-                .experimentItems(getExperimentItems(row.get("experiment_items_array", List[].class)))
-                .lastUpdatedAt(row.get("last_updated_at", Instant.class))
-                .createdAt(row.get("created_at", Instant.class))
-                .createdBy(row.get("created_by", String.class))
-                .lastUpdatedBy(row.get("last_updated_by", String.class))
-                .build());
+        return results.map((row, rowMetadata) -> {
+
+            Map<String, JsonNode> data = getData(row);
+
+            JsonNode input = getJsonNode(row, data, "input");
+            JsonNode expectedOutput = getJsonNode(row, data, "expected_output");
+            JsonNode metadata = getJsonNode(row, data, "metadata");
+
+            return DatasetItem.builder()
+                    .id(row.get("id", UUID.class))
+                    .input(input)
+                    .data(data)
+                    .expectedOutput(expectedOutput)
+                    .metadata(metadata)
+                    .source(DatasetItemSource.fromString(row.get("source", String.class)))
+                    .traceId(Optional.ofNullable(row.get("trace_id", String.class))
+                            .filter(s -> !s.isBlank())
+                            .map(UUID::fromString)
+                            .orElse(null))
+                    .spanId(Optional.ofNullable(row.get("span_id", String.class))
+                            .filter(s -> !s.isBlank())
+                            .map(UUID::fromString)
+                            .orElse(null))
+                    .experimentItems(getExperimentItems(row.get("experiment_items_array", List[].class)))
+                    .lastUpdatedAt(row.get("last_updated_at", Instant.class))
+                    .createdAt(row.get("created_at", Instant.class))
+                    .createdBy(row.get("created_by", String.class))
+                    .lastUpdatedBy(row.get("last_updated_by", String.class))
+                    .build();
+        });
+    }
+
+    private Map<String, JsonNode> getData(Row row) {
+        return Optional.ofNullable(row.get("data", Map.class))
+                .filter(s -> !s.isEmpty())
+                .map(value -> (Map<String, String>) value)
+                .stream()
+                .map(Map::entrySet)
+                .flatMap(Collection::stream)
+                .map(entry -> Map.entry(entry.getKey(), getJsonNodeFromString(entry.getValue())))
+                .collect(toMap(Map.Entry::getKey, Map.Entry::getValue));
+    }
+
+    private JsonNode getJsonNodeFromString(String value) {
+        var data = JsonUtils.readValue(value, new TypeReference<DatasetItemData>() {
+        });
+
+        return data.value();
```

Comment:
This DAO is doing the whole effort of resolving the type of the JSON elements, but then is never using them. This retrieval doesn't resolve them and the columns field isn't providing it neither as discussed in the ticket.

Reply:
If I understood correctly, the types would be used in the future.

Reply:
Agreed to not to store this data at the moment.